netbox/extras/views.py

```diff
@@ -357,31 +367,54 @@ def get(self, request):
         })
 
 
-class ReportView(ObjectPermissionRequiredMixin, View):
+class GetReportMixin:
+    def get_report(self, name):
+        if '.' not in name:
+            raise Http404
+        # Retrieve the Report by "<module>.<report>"
+        module_name, report_name = name.split('.', 1)
```

Comment:
My only gripe with this is the inconsistency with the API routes. Any thoughts on if we should change those?

Reply:
I think the API stuff is fine; the API ViewSets are designed to take a single PK anyway. I just mean changing

```python
path('reports/<str:name>/', views.ReportView.as_view(), name='report'),
```

to

```python
path('reports/<str:module>.<str:name>/', views.ReportView.as_view(), name='report'),
```

similar to what we have now for scripts.

Reply:
For scripts today we have:
```
path('scripts/<str:module>/<str:name>/', views.ScriptView.as_view(), name='script'),
```
which is annoyingly matching `scripts/results/123/`. That is easily overcome by moving the results path above the standard view, but is a bit confusing. Would you be okay with changing scripts (and reports to)?:

```
path('scripts/<str:module>.<str:name>/', views.ScriptView.as_view(), name='script'),
```
I like the dot because it is consistent with the API and we have that sort of consistently in most other places.